### PR TITLE
Create tankbuilder class for unit testing

### PR DESCRIPTION
**Pull request by BryanFriestad**

### CODE REVIEW

File: src/test/java/pro/trevor/tankgame/util/TankBuilder.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        return this;
+    }
+
+    public TankBuilder<T, E> atPosition(Position p)
+    {
+        tank.setPosition(p);
+        return this;
+    }
+
+    public T finish()
```
Comment (TrevorBrunette) [inline src/test/java/pro/trevor/tankgame/util/TankBuilder.java:31]: I would name this `build`

File: src/test/java/pro/trevor/tankgame/util/TankBuilder.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+public class TankBuilder<T extends GenericTank<E>, E extends Enum<E> & IAttribute> 
+{
+    private final T tank;
+    public TankBuilder(T tank) 
+    {
+        this.tank = tank;
+    }
+
+    public TankBuilder<T, E> withAttribute(E attribute, Object value)
```
Comment (TrevorBrunette) [inline src/test/java/pro/trevor/tankgame/util/TankBuilder.java:19]: `withAttribute` is probably fine, but we might want to use `with` to make chaining less verbose.

File: src/test/java/pro/trevor/tankgame/util/TankBuilder.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import org.json.JSONObject;
+
+import pro.trevor.tankgame.rule.impl.version3.Tank;
+import pro.trevor.tankgame.rule.impl.version3.TankAttribute;
+import pro.trevor.tankgame.state.board.Position;
+import pro.trevor.tankgame.state.board.attribute.IAttribute;
+import pro.trevor.tankgame.state.board.unit.GenericTank;
+
+public class TankBuilder<T extends GenericTank<E>, E extends Enum<E> & IAttribute> 
+{
```
Comment (TrevorBrunette) [inline src/test/java/pro/trevor/tankgame/util/TankBuilder.java:12]: open braces should be on the same line as the function or class declaration

File: src/test/java/pro/trevor/tankgame/util/TankBuilder.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        this.tank = tank;
+    }
+
+    public TankBuilder<T, E> withAttribute(E attribute, Object value)
+    {
+        tank.set(attribute, value);
+        return this;
+    }
+
+    public TankBuilder<T, E> atPosition(Position p)
```
Comment (TrevorBrunette) [inline src/test/java/pro/trevor/tankgame/util/TankBuilder.java:25]: `atPosition` is probably fine, but we might want to use `at` to make chaining less verbose.

**Review by TrevorBrunette — CHANGES_REQUESTED** [review]

File: src/test/java/pro/trevor/tankgame/util/TankBuilder.java (hunk 1 above)
Comment (BryanFriestad) [reply src/test/java/pro/trevor/tankgame/util/TankBuilder.java:31]: Right, but I wouldn't, so I didn't.

Comment (BryanFriestad) [reply src/test/java/pro/trevor/tankgame/util/TankBuilder.java:31]: It starts with "build...()", so ending with something different just makes sense.

File: src/test/java/pro/trevor/tankgame/util/TankBuilder.java (hunk 2 above)
Comment (BryanFriestad) [reply src/test/java/pro/trevor/tankgame/util/TankBuilder.java:19]: ACK

File: src/test/java/pro/trevor/tankgame/util/TankBuilder.java (hunk 4 above)
Comment (BryanFriestad) [reply src/test/java/pro/trevor/tankgame/util/TankBuilder.java:25]: ACK

File: src/test/java/pro/trevor/tankgame/util/TankBuilder.java (hunk 3 above)
Comment (BryanFriestad) [reply src/test/java/pro/trevor/tankgame/util/TankBuilder.java:12]: ACK

**Review by TrevorBrunette — APPROVED** [review]

**Pull request merged**